### PR TITLE
libbs2b: fix -flat_namespace usage

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
This formula is blocked from bottling on Monterey by a `-flat_namespace` audit failure. Our standard patches don't work with the configure script here, so I just added a call to `autoreconf`.

File: Formula/libbs2b.rb [hunk 1]
```diff
@@ -21,7 +21,18 @@ class Libbs2b < Formula
...
 
+  on_macos do
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+  end
+
   def install
+    if OS.mac?
+      inreplace "configure.ac", "dist-lzma", ""
```
Comment (carlocab) [inline Formula/libbs2b.rb:32]: Why do we need this?

Comment (danielnachun) [reply Formula/libbs2b.rb:32]: Ah I forgot to commit my comment!  It's pretty simple - `autoconf` no longer allows `dist-lzma` in `configure.ac` and fails if this isn't removed.  I'll add the comment so this is more clear.

**Comment by danielnachun** [comment]
I added a comment to explain why that `inreplace` is needed, and updated this to use the long names for the `autoreconf` arguments.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**